### DISCUSSION TITLE
How do I programmatically change the the run_name from a step in my workflow?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#53631] [Actions] [answered]
### Select Topic Area

Question

### Body

We don't deploy every build. Either way I wanted to give people a way to do a manual deploy of any build (and our builds are based on the commit SHA). Now they might just deploy the last build of the branch and great if they do, the don't have to tell the workflow anything, they just run it in the context of that branch and I'm building all the branches so that works. However they might want to push out a different build of that same branch... maybe a PR build .. and I want to make things easy so I have a workflow that opens like this :

```
name: deploy
run-name: Deploy ${{ inputs.environment }}
on:
  workflow_dispatch:
    inputs:
      ENVIRONMENT:
        type: environment
        description: To Environment
      SHA:
        type: string
        description: Artifact SHA (not required)
jobs:
  info:
    name: Determine Info
    runs-on: ubuntu-latest
    steps:
      - name: Determine SHA
        id: identify
        run: |
          if [ "${{inputs.SHA}}X" == "X" ]; then
            export GIT_SHA="${{ github.sha }}"
          else
            export GIT_SHA="${{inputs.SHA}}"
          fi
       # how do I update the run name here to be : Deploy ${{ inputs.environment }} - ${GIT_SHA} ? ... thank you !
```

The user can optionally tell me the SHA (which is the build id in my situation) and what I want to do is be able to update that workflow run_name to include the variable GIT_SHA. Is there a way to do this in my "Determine SHA" step ?

**Comment by ChristopherHX** [comment]
You cannot update the run-name as of now.

Rewrite your `Determine SHA` to not depend on bash, you are not doing anything which wouldn't work directly in Action expression.

example
```yaml
run-name: Deploy ${{ inputs.environment }} - ${{ inputs.SHA || github.sha }}
```

`inputs.SHA || github.sha` the value is `inputs.SHA` if it is a non empty string otherwise `github.sha`

See documentation https://docs.github.com/en/actions/learn-github-actions/expressions#operators

**Reply by ghost** [reply, marked as answer]
I think that seemed to work ...

The 1st time I ran with input.SHA as 9a7ed23255a3c6ab026e747b12aa4b1d4b5e3dad 
The 2nd time I ran the workflow without specifying any input.SHA and it picked up the latest SHA of the branch that I ran the workflow from.

<img width="763" alt="Screenshot 2023-04-25 at 6 00 25 PM" src="https://user-images.githubusercontent.com/69161124/234413909-a2b78b5a-74c4-4915-a2c8-b382a79caaaf.png">

Thank you .. I didn't realize I could put expressions in the run_name 👍

**Reply by eloisetaylor5693** [reply]
Is it still not possible to set during the run?

I'm looking to do something similar, but keep the commit message as the run name if triggered by a push, and a different message if not

**Reply by ixti** [reply]
Likewise, but I want to update run name with tag that was generated :D

**Reply by vgrafe** [reply]
I have the same use case than @ixti, wonder if it can be done.

**Reply by orlovmyk** [reply]
@vgrafe Looks like it can't be done
From docs:
>This value can include expressions and can reference the [github](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) and [inputs](https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context) contexts.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name

**Comment by koder101** [comment]
Also sharing this workflow if someone was looking to include the environment name dynamically to `run-name`.

I can confirm that this is working for me. Although the doc says that the `run-name` has only access to `{{ inputs }}` and `{{ github }}` contexts, but when I added a variable at the repository level (under Github secrets) and accessed it using the `vars.DEFAULT_ENVIRONMENT` surprisingly it worked. You may use the same trick to achieve what you are looking for.

May be this is an undocumented feature ( _bit reluctant to post if they remove it after its discovery_ 😁 ).

```
on:
  push:
    branches:
      - main
    paths:
      - 'abcd/**'
  workflow_dispatch:
    inputs:
      environment:
        description: 'Choose the deployment environment'
        required: false
        default: 'DEV'
        type: choice
        options:
          - DEV
          - UAT
          - PROD

run-name: '${{ github.event.inputs.environment || vars.DEFAULT_ENVIRONMENT }} - Run ID: ${{ github.run_id }} - by @${{ github.actor }}'
```
